src/box/lua/config/cluster_config.lua

```diff
@@ -205,6 +205,15 @@ local nested_cluster_config = schema.new(schema_name, record_from_fields({
                         value = instance_config_with_scope('instance'),
                     }),
                 }),
+                validate = function(replicasets, w)
```

Comment:
We discussed, whether it is possible that the same configuration mistake may appear in the upper scopes: empty `replicasets`, empty `groups`.

We also discussed that a partial config may miss `groups`. For example, a local file config may contain global options to connect to etcd or tarantool based configuration storage, while all the groups/replicasets/instances are in the remote config storage.

Another similar case is when a key in a remote storage contains just global options.

The main point is that a presence of `groups` is validated on the whole configuration, compiled from all the sources/keys, not before.

Whether we should apply the same logic for `replicasets` and `instances`?

For example, one may want to define options for a replicaset:

```yaml
groups:
  g-001:
    replicasets:
      r-001:
        replication:
          failover: election
        memtx:
          memory = <...>
        labels:
          size: large
```

But have `instances` defined in another key.

```yaml
groups:
  g-001:
    replicasets:
      r-001:
        instances:
          i-001: {}
          i-002: {}
          i-003: {}
```

Does it look meaningful?

Reply:
We've discussed it more thoroughly and decided to forbid only empty replicasets and groups in merged config.

Reply:
Fixed.

Reply:
Thank you!